include/ptp_clock.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2017 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PTP_CLOCK_H_
+#define _PTP_CLOCK_H_
+
+#include <stdint.h>
+#include <device.h>
+#include <misc/util.h>
+#include <net/gptp.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*
+ * Special alignment is needed for ptp_clock which is stored in
+ * a ptp_clock linker section if there are more than one ptp clock
+ * in the system. If there is only one ptp clock,
+ * then this alignment is not needed, unfortunately this cannot be
+ * known beforehand.
+ *
+ * The ptp_clock struct needs to be aligned to 32 byte boundary,
+ * otherwise the __ptp_clock_end will point to wrong location and ptp_clock
+ * initialization done in ptp_clock_init() will not find proper values
+ * for the second interface.
+ *
+ * So this alignment is a workaround and should eventually be removed.
+ */
+#define __ptp_clock_align __aligned(32)
+
+struct ptp_clock;
+
+struct ptp_clock_driver_api {
+	int (*set)(struct ptp_clock *clk, struct net_ptp_time *tm);
+	int (*get)(struct ptp_clock *clk, struct net_ptp_time *tm);
+	int (*adjust)(struct ptp_clock *clk, int increment);
+	int (*rate_adjust)(struct ptp_clock *clk, float ratio);
+};
+
+/**
+ * @brief PTP Clock structure
+ *
+ * Used to handle a ptp clock on top of a device driver instance.
+ * There can be many ptp_clock instance against the same device.
+ */
+struct ptp_clock {
+	/** The actually device driver instance the ptp_clock is related to */
+	struct device *dev;
+
+	/** API for the ptp clock. */
+	struct ptp_clock_driver_api const *api;
+} __ptp_clock_align;
+
+#define PTP_CLOCK_GET_NAME(dev_name, sfx) (__ptp_clock_##dev_name##_##sfx)
```

Comment:
Instead of 'sfx', could this be clearer as 'clock_name'?

And as 'struct device' is needed in the end, is it clearer to use a device name or an actual struct device here?

Reply:
sfx is used in all other places with similar macros so I think we should stick with same convention.